cmk/notification_plugins/signl4.py

```diff
@@ -24,24 +26,46 @@ def _signl4_url() -> str:
 
 
 def _signl4_msg(context: dict[str, str]) -> dict[str, object]:
-    host_name = context["HOSTNAME"]
-    notification_type = context["NOTIFICATIONTYPE"]
+    host_name = context.get("HOSTNAME", "")
+    service_desc = context.get("SERVICEDESC", "")
+    host_state = ""
+    notification_type = context.get("NOTIFICATIONTYPE", "")
     host_problem_id = context.get("HOSTPROBLEMID", "")
     service_problem_id = context.get("SERVICEPROBLEMID", "")
+    description = f"{notification_type} on {host_name}"
 
+    # Prepare description information
+    if context.get("WHAT", "") == "SERVICE":
+        if notification_type in ["PROBLEM", "RECOVERY"]:
+            description += " (" + service_desc + ")"
+        else:
+            description += " (" + service_desc + ")"
+    else:
+        if notification_type in ["PROBLEM", "RECOVERY"]:
+            host_state = context.get("HOSTSTATE", "") or ""
+            description += " (" + host_state + ")"
+        else:
+            description += " (" + host_state + ")"
     # Remove placeholder "$SERVICEPROBLEMID$" if exists
     if service_problem_id.find("$") != -1:
         service_problem_id = ""
-
     # Check if this is a new problem or a recovery
     s4_status = "new" if notification_type != "RECOVERY" else "resolved"
 
+    # Base64 encode the SERVICEDESC for matching updates for service alerts
```

Comment:
Is the servicedescription enough to identify the correct matching on signl4?
What about multiple hosts/sites with the same service description?

Reply:
Thanks and good question. I was looking for something like a service ID. The service descriptions seemed to be the most unique choice. Do you have any other idea or best practice?

Reply:
host+service+site / host+site should be unique, you can get the site with 

`context["OMD_SITE"]`